Code/Common/include/Ancillary/type_list2.h

```diff
@@ -0,0 +1,255 @@
+/*=========================================================================
+*
+*  Copyright NumFOCUS
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*         http://www.apache.org/licenses/LICENSE-2.0.txt
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+*=========================================================================*/
+#ifndef sitk_typelist_h_
+#define sitk_typelist_h_
+
+#include <tuple>
+#include <type_traits>
+#include <algorithm>
+
+namespace typelist2
+{
+
+/** \struct typelist
+ * \brief an non-instantiable holder of variadic template parameters to represent a list of types
+ *
+ * Can be used as a not and not an object. It enables complex compile-time manipulation of sets of types.
+ */
+template <typename... Ts>
+struct typelist;
+
+/** \brief Provides access to the number of types in a typelist.
+ *
+ * Uses std::integral_constant to define member type "value_type" and member constant "value".
+ */
+template <typename Typelist>
```

Comment:
I think the ITK code convention would be to have an extra `T` in front of names of template arguments, like `TTypelist`, or `TTypeList`. But it's up to you, of course.

Reply:
Good suggestion.